packages/slate-react/src/components/content.js

```diff
@@ -192,14 +197,23 @@ class Content extends React.Component {
 
     // Otherwise, set the `isUpdatingSelection` flag and update the selection.
     this.tmp.isUpdatingSelection = true
-    native.removeAllRanges()
-
-    // COMPAT: Again, since the DOM range has no concept of backwards/forwards
-    // we need to check and do the right thing here.
-    if (isBackward) {
-      native.setBaseAndExtent(endContainer, endOffset, startContainer, startOffset)
-    } else {
-      native.setBaseAndExtent(startContainer, startOffset, endContainer, endOffset)
+
+    // COMPAT: IE 11 does not support Selection.extend
+    if (native.extend) {
+      native.removeAllRanges()
+      // COMPAT: Since the DOM range has no concept of backwards/forwards
+      // we need to check and do the right thing here.
+      if (isBackward) {
+        native.collapse(range.endContainer, range.endOffset)
+        native.extend(range.startContainer, range.startOffset)
+      } else {
+        native.collapse(range.startContainer, range.startOffset)
+        native.extend(range.endContainer, range.endOffset)
+      }
+    } else if (selection.isCollapsed) {
```

Comment:
I'm not sure about this check, because won't this mean that in IE11 when the selection is expanded we won't be updating it at all? This seems like it won't handle programmatic selection changes that result in an expanded selection.

Reply:
it's because I introduced a weird bug in IE with the previous change (by checking the direction and the focus node, it's not necessary anymore) 
change is on the way